Adding libraries for holland

marcoamorales opened the pull request.

Commit:
initial library skeleton

Commit:
Update mysqldump recipe to work with new mysql community cookbook

marcoamorales committed:
action install written, trying testing for it

marcoamorales committed:
fixed resource name error & misc errors

marcoamorales committed:
added delete action to hollandbackup_installer

Commit:
Splitting library tests into two recipes

Commit:
Adding serverspec tests for install and uninstall

Commit:
Updating directory tests for epel installs

Commit:
Bumping version of CentOS to test

Commit:
Adding 14.04 to test kitchen

marcoamorales committed:
added hwrp for hollandbackup_main

marcoamorales committed:
added hwrp for hollandbackup_mysqldump

marcoamorales committed:
added hwrp for backupsets

martinb3 commented:
Looks to be 70+ rubocop offenses detected, 4 foodcritic offenses.

martinb3 commented on libraries/provider_hollandbackup_main.rb:
```diff
@@ -0,0 +1,27 @@
+
+class Chef
+  class Provider
+    class HollandbackupMain < Chef::Provider::LWRPBase
```
It looks like you have many of these providers and resources defined twice -- once under `libraries/...` and once in `resources/...` & `providers/...`. They really should just be in one of those places, especially if they're identical.

martinb3 commented on libraries/provider_hollandbackup_mysqldump.rb:
```diff
@@ -0,0 +1,26 @@
+      use_inline_resources if defined?(use_inline_resources)
+
+      action :configure do
+        template 'holland.conf' do
```
This appears to be the same template for both 'main' and 'mysqldump' -- could these just be one provider with an attribute for a toggle? It would reduce the duplicated files down to a single attribute (much shorter).

martinb3 commented on libraries/resource_hollandbackup_install.rb:
```diff
@@ -0,0 +1,19 @@
+      actions :create, :delete
+      default_action :create
+
+      attribute :repo_name, kind_of: String, default: nil
```
I think the default is already nil on these if you don't supply one. Ditto for all the other resources defined.

martinb3 commented on libraries/provider_hollandbackup_install.rb:
```diff
@@ -0,0 +1,75 @@
+      end
+
+      action :delete do
+        # TODO: Delete the gpg key
```
Do we need this TODO done before release?

martinb3 commented:
`hollandbackup_main` feels a little wordy. Could we combine `hollandbackup_main` and `hollandbackup_mysqldump` into one resource (and potentially one provider, too), and just call it hollandbackup?

martinb3 commented on libraries/resource_hollandbackup_backupset.rb:
```diff
@@ -0,0 +1,68 @@
+      default_action :configure
+
+      # template
+      attribute :name, :kind_of => String, :name_attribute => true
```
Some of these appear to be duplicated across different resources. Consider a shared parent class, or use poise's helper for template content - https://github.com/poise/poise#template-content.

martinb3 commented:
Two of the providers appear to be a single template file. Would it make sense to combine them into a single provider with an attribute to toggle the file being templated? Or at least figure out ways to share more of the provider and resource code (like a parent class the resources and/or providers share).

martinb3 commented:
I'm also seeing serverspec tests fail on `library-install-centos-65`:

```
       Finished in 0.52007 seconds (files took 0.45521 seconds to load)
       22 examples, 7 failures

       Failed examples:

       rspec /tmp/busser/suites/serverspec/holland_install_spec.rb:33 # File "/etc/holland/backupsets/default.conf" should be file
       rspec /tmp/busser/suites/serverspec/holland_install_spec.rb:34 # File "/etc/holland/backupsets/default.conf" should be owned by "root"
       rspec /tmp/busser/suites/serverspec/holland_install_spec.rb:35 # File "/etc/holland/backupsets/default.conf" should be grouped into "root"
       rspec /tmp/busser/suites/serverspec/holland_install_spec.rb:36 # File "/etc/holland/backupsets/default.conf" should be mode 644
       rspec /tmp/busser/suites/serverspec/holland_install_spec.rb:37 # File "/etc/holland/backupsets/default.conf" content should match /\[mysqldump\]/
       rspec /tmp/busser/suites/serverspec/holland_install_spec.rb:41 # Command "holland bk" exit_status should eq 0  
```

marcoamorales committed:
Deprecating recipes and LWRPs.

Rewriting this cookbook to leverage HWRPs will require to deprecate the
old resources and providers to avoid any namespacing problems.

I'm also renaming the test cookbook and suites in kitchen.

marcoamorales committed:
rubocop fixes

marcoamorales committed:
Fix hollandbackup_backupsets and errors on centos

marcoamorales committed:
collapse _install and _main into one single hwrp
